### PR TITLE
Verification script

### LINKED ISSUE #5: Invoke plugin verification script during build
GlooE now publishes a script to verify plugin compatibility. Download the script and call it to verify the compiled plugin files.

### PR DESCRIPTION
**Pull request by marcogschmidt**
BOT NOTES: 
resolves https://github.com/solo-io/ext-auth-plugins/issues/5

### CODE REVIEW

File: changelog/v0.0.2/verify-plugins-script.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changelog:
+- type: NEW_FEATURE
+  description: Verify plugin compatibilioty with GlooE during build.
```
Comment (kdorosh) [inline changelog/v0.0.2/verify-plugins-script.yaml]: nit: typo 
```suggestion
  description: Verify plugin compatibility with GlooE during build.
```

File: examples/required_header/pkg/impl.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+func (p *RequiredHeaderPlugin) NewConfigInstance(ctx context.Context) (interface{}, error) {
+	return &Config{}, nil
+}
+
+func (p *RequiredHeaderPlugin) GetAuthService(ctx context.Context, configInstance interface{}) (api.AuthService, error) {
+	config, ok := configInstance.(*Config)
+	if !ok {
+		return nil, errors.New(fmt.Sprintf("unexpected config type %T", configInstance))
```
Comment (kdorosh) [inline examples/required_header/pkg/impl.go]: nit: define this error at the top of the file

File: Makefile [hunk 3]
```diff
@@ -13,41 +7,67 @@ ifeq ($(TAGGED_VERSION),)
...
 VERSION ?= $(shell echo $(TAGGED_VERSION) | cut -c 2-)
 
-.PHONY: compare-deps
-compare-deps: Gopkg.lock GlooE-Gopkg.lock print-info
-	go run scripts/compare_dependencies.go Gopkg.lock GlooE-Gopkg.lock
+#----------------------------------------------------------------------------------
+# Retrieve GlooE build information
+#----------------------------------------------------------------------------------
+GLOOE_DIR := _glooe
+z := $(shell mkdir -p $(GLOOE_DIR))
```
Comment (kdorosh) [inline Makefile]: is `z` used anywhere?

Comment (marcogschmidt) [reply Makefile]: No, it's a quick way of creating the dir. Do you know if I can use `_`?

Comment (marcogschmidt) [reply Makefile]: It works, will change.

File: Makefile [hunk 4]
```diff
@@ -13,41 +7,67 @@ ifeq ($(TAGGED_VERSION),)
...
+#----------------------------------------------------------------------------------
+# Build plugins
 #----------------------------------------------------------------------------------
 EXAMPLES_DIR := examples
 SOURCES := $(shell find . -name "*.go" | grep -v test)
 
-.PHONY: publish-examples
-publish-examples:
+define get_glooe_var
+$(shell grep $(1) $(GLOOE_DIR)/build_env | cut -d '=' -f2)
```
Comment (kdorosh) [inline Makefile]: feels more natural to use awk here

i.e.,
```
awk -F"=" '{print $2}' $(GLOOE_DIR)/build_env
```
you can combine that with piping to grep, but awk syntax is expressive enough you can get what you want without grep. would have to google the syntax myself. this is a nit though feel free to ignore

Comment (kdorosh) [reply Makefile]: I believe the syntax without grep, cut, or any piping is:
```
awk -F"=" '/$(1)/ {print $2}' $(GLOOE_DIR)/build_env
```

Comment (marcogschmidt) [reply Makefile]: Not very familiar with `awk`. I tried the command above and it seems to range over all the lines in the file and returning newlines. I did a couple of google searches but could not find a way of matching only one line. I would leave it like this for now, feel free to update if you want.

image

Comment (marcogschmidt) [reply Makefile]: Tried:
```
awk -F"=" '/$(1)/ {print $2}' $(GLOOE_DIR)/build_env
```
but somehow make does not like the $(1) being quoted... I'll leave it like it is for now.

File: Makefile [hunk 5]
```diff
@@ -13,41 +7,67 @@ ifeq ($(TAGGED_VERSION),)
...
-publish-examples:
+define get_glooe_var
+$(shell grep $(1) $(GLOOE_DIR)/build_env | cut -d '=' -f 2-)
+endef
+
+.PHONY: build-plugins
+build-plugins: $(GLOOE_DIR)/build_env $(GLOOE_DIR)/verify-plugins-linux-amd64
+	docker build --no-cache -t quay.io/solo-io/ext-auth-plugins:$(VERSION) \
+		--build-arg GO_BUILD_IMAGE=$(call get_glooe_var,GO_BUILD_IMAGE) \
+		--build-arg GC_FLAGS=$(call get_glooe_var,GC_FLAGS) \
```
Comment (kdorosh) [inline Makefile:52]: looks like we're hard-coding all the possible values in `build_env` here. what if we wanted to add a new `FLAGS` build arg that isn't for GC, or we wanted to split out `GO_BUILD_IMAGE` into its constituent parts? Seems like we should just take all the env vars in the `build_env` we download and pass them through to the container

**Review by kdorosh — COMMENTED** [review]
quick improvement to make the makefile more future-proof, but otherwise lgtm 👍

File: Makefile (hunk 5 above)
Comment (kdorosh) [reply Makefile:52]: Ah, the dockerfile using these variables is also in this repo, so I feel less strongly about this now

Comment (kdorosh) [reply Makefile:52]: LGTM 👍

**Review by kdorosh — APPROVED** [review]

**Pull request merged**